openpype/hosts/houdini/plugins/publish/collect_cache_farm.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pyblish.api
+import hou
+from openpype.hosts.houdini.api import lib, colorspace
+from openpype.hosts.houdini.api.lib import get_color_management_preferences
+
+
+class CollectDataforCache(pyblish.api.InstancePlugin):
+    """Collect data for caching to Deadline."""
+
+    order = pyblish.api.CollectorOrder + 0.04
+    families = ["ass", "pointcache",
+                "mantraifd", "redshiftproxy"]
+    hosts = ["houdini"]
+    targets = ["local", "remote"]
+    label = "Collect Data for Cache"
+    chunkSize = 999999
+
+    def process(self, instance):
+        if instance.data.get("farm", True):
+            for family in instance.data["families"]:
+                instance.data["family"] = family
+                if family == "bgeo" or "abc":
+                    instance.data["family"] = "pointcache"
+                instance.data["subset"] = instance.name
+            # instance.data["expected_families"]
+            # only created for farm caching
+            instance.data["expected_families"] = list()
+
+            instance.data["expected_families"].append(
+                instance.data["family"])
+            instance.data["families"].append("publish.hou")
+            data = dict()
+
+            data["plugin"] = "Houdini"
+            data["publish"] = True
+            ropnode = hou.node(instance.data["instance_node"])
+            instance.data.update(data)
+            output_parm = lib.get_output_parameter(ropnode)
+            # TODO: make sure the expectedfiles include all the data
+            expected_filepath = output_parm.eval()
+            if "files" not in instance.data:
+                instance.data["files"] = list()
+            if "expectedFiles" not in instance.data:
+                instance.data["expectedFiles"] = list()
+            cache_files = dict()
+            if instance.data.get("frames"):
+                files = self.get_files(
+                    instance, expected_filepath)
+                # list of files
+                instance.data["files"].extend(files)
+            else:
+                instance.data["files"].append(output_parm.eval())
+            cache_files = {"_": instance.data["files"]}
+            instance.data["expectedFiles"].append(cache_files)
+
+            # update the colorspace data
+            colorspace_data = get_color_management_preferences()
+            instance.data["renderProducts"] = colorspace.ARenderProduct()
```

Comment:
Where is this used?

Also, if this is needed - I suspect it'd need to be a list of render products?
And why is `ARenderProduct` in the `colorspace` lib? :)

Reply:
If there is no instance.data["renderProducts"], the submit publish job would give KeyError if that instance

Reply:
I think we're mixing up two things too much - the **SubmitPublishJob** is about creating an additional 'publish job'  which publishes rendered outputs (renders/image sequences) from a host DCC after the render job itself has finished. The docstring is also quite specific about that it is for render jobs / image sequences.

There's assembly logic (for tiled rendering), there's aov logic, multiple cameras, multiple "separate publish jobs" from a single instance, etc. I don't see exactly why're ending up with quite a convoluted implementation as we currently are to publish instances.

I feel like 50% of the logic in this PR is about somehow 'wrestling' your new instances into that plug-in so that you can just get something as simple as a pointcache published on the farm.

Reply:
> I think we're mixing up two things too much - the **SubmitPublishJob** is about creating an additional 'publish job' which publishes rendered outputs (renders/image sequences) from a host DCC after the render job itself has finished. The docstring is also quite specific about that it is for render jobs / image sequences.
> 
> There's assembly logic (for tiled rendering), there's aov logic, multiple cameras, multiple "separate publish jobs" from a single instance, etc. I don't see exactly why're ending up with quite a convoluted implementation as we currently are to publish instances.
> 
> I feel like 50% of the logic in this PR is about somehow 'wrestling' your new instances into that plug-in so that you can just get something as simple as a pointcache published on the farm.

do you mean we need to have another submit publish job for farm cache?

Reply:
If that means we'll have to jump through 100 less hoops to get this feature I'd say so, yes. I feel like you're trying to wrangle your way through something that was never designed to do what you want it to do.